builtin/providers/aws/resource_aws_kinesis_analytics.go

```diff
@@ -0,0 +1,541 @@
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"record_row_delimiter": { //MappingParameters.CSVMappingParameters.RecordRowDelimiter
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"record_column_delimiter": { //MappingParameters.CSVMappingParameters.RecordColumnDelimiter
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+
+						"columns": { //InputSchema.RecordColumns
+							Type:     schema.TypeSet,
+							Required: true,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"sql_type": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"mapping": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+
+			"outputs": {
+				Type:     schema.TypeSet,
+				Optional: true,
+				ForceNew: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": { // Name
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"record_format_type": { // DestinationSchema.RecordFormatType ( JSON || CSV)
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"arn": { // KinesisStreamsOutput.ResourceARN || KinesisFirehoseOutput.ResourceARN
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"role_arn": { // KinesisStreamsOutput.RoleARN || KinesisFirehoseOutput.RoleARN
+							Type:     schema.TypeString,
+							Required: true,
+						},
+					},
+				},
+			},
+
+			"arn": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"version_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+
+			"create_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func discoverStreamType(arn string) (string, error) {
+
+	streamType := ""
+
+	if strings.Contains(arn, "aws:firehose") {
+		streamType = "firehose"
+	} else if strings.Contains(arn, "aws:kinesis") {
+		streamType = "kinesis"
+	}
+
+	if streamType == "" {
+		return "", fmt.Errorf("Error when attempting to determine stream type from AWS ARN. The arn (%s), does not appear to be kinesis nor firehose", arn)
+	}
+
+	return streamType, nil
+}
+
+func createOutputList(outputs []interface{}) ([]*kinesisanalytics.Output, error) {
+
+	var outputStreams []*kinesisanalytics.Output
+
+	for _, elem := range outputs {
+
+		output := elem.(map[string]interface{})
+
+		streamType, err := discoverStreamType(output["arn"].(string))
+
+		if err != nil {
+			return nil, err
+		}
+
+		if streamType == "kinesis" {
+			outputStreams = append(outputStreams, &kinesisanalytics.Output{
+
+				Name: aws.String(output["name"].(string)),
+
```

Comment:
If these are optional values, we could potentially get panic's here - we may need to check that the Map actually has a property named `name`, `record_format_type` etc